### ISSUE TITLE
ckeygen --showpub fails with a TypeError

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5889]
|<img alt="ltaylor.volks's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ltaylor.volks reported|
|-|-|
|Trac ID|trac#5889|
|Type|defect|
|Created|2012-08-19 23:17:34Z|

The command:

```
ckeygen --showpub --filename=/Users/blah/.ssh/id_rsa
```

should display the public key, but throws a TypeError

This call in `twisted.conch.scripts.ckeygen.displayPublicKey` omits the 'type' argument of toString():  

```
print keys.Key(key).public().toString()
```

Valid types can be one of "OPENSSH", "LSH", "AGENTV3"

`twisted.conch.scripts.ckeygen.py` should either provide a default format when calling toString() (OPENSSH?), or allow the user to specify the desired output format.

Attachments:

* [5589-ckeygen](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2f7/2f7bc0048f5b29dbb7fb1264a45ef898562bd0d1/1e4d783772ebbde3ac6f89ab22e3a6dfeebf9716) (1766 bytes) - added by ltaylor.volks on 2012-08-20 00:13:31Z - Patch to provide 'openssh' arg, with test
* [5889.bugfix](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2f7/2f7bc0048f5b29dbb7fb1264a45ef898562bd0d1/85d73afb048faa05a6554d9194ae6063a65bf582.bugfix) (78 bytes) - added by ltaylor.volks on 2012-08-20 05:10:59Z - Added topfile NEWS entry
* [5889-ckeygen.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2f7/2f7bc0048f5b29dbb7fb1264a45ef898562bd0d1/f4ebf29e9c636b15072354aaa9b64251d89e7c89.diff) (1762 bytes) - added by ltaylor.volks on 2012-08-21 00:48:56Z - Patch to provide 'openssh' arg, with test (SVN diff...ignore previous attachment)

<details><summary>Searchable metadata</summary>

```
trac-id__5889 5889
type__defect defect
reporter__ltaylor_volks ltaylor.volks
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__easy easy
time__1345418254000000 1345418254000000
changetime__1346148869000000 1346148869000000
version__None None
owner__therve therve
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @therve|
|-|-|

Looks good! I'm going to merge it along [#5890](https://github.com/twisted/twisted/issues/5890).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [35420]) Fix ckeygen --showpub behavior, passing a type to display in toString and
catching the proper exception for encrypted keys.

Author: ltaylor.volks
Reviewer: therve
Fixes: [#5889](https://github.com/twisted/twisted/issues/5889), [#5890](https://github.com/twisted/twisted/issues/5890)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

This patch appears to misuse `str.strip`.  Consider:

```
>>> 'hello'.strip(' comment')
'hell'
>>> 
```

I think this new code makes no sense:

```
    self.assertEqual( 
        self.stdout.getvalue().strip('\n'), 
        publicRSA_openssh.strip(' comment'))
```

Please file a ticket for correcting this.  Thanks!
